pkg/subctl/cmd/show_gateways.go

```diff
@@ -0,0 +1,99 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	submarinerclientset "github.com/submariner-io/submariner-operator/pkg/client/clientset/versioned"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/submarinercr"
+	submv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type gatewaysStatus struct {
+	node        string
+	haStatus    submv1.HAStatus
+	connections string
+	summary     string
+}
+
+func newGatewayStatusFrom(node string, haStatus submv1.HAStatus, connections string, summary string) gatewaysStatus {
+	v := gatewaysStatus{
+		node:        node,
+		haStatus:    haStatus,
+		connections: connections,
+		summary:     summary,
+	}
+	return v
+}
+
+var showGatewaysCmd = &cobra.Command{
+	Use:   "gateways",
+	Short: "Get information on your gateways related to submariner",
+	Long:  `This command shows the status of submariner gateways in your cluster.`,
+	Run:   showGateways,
+}
+
+func init() {
+	showCmd.AddCommand(showGatewaysCmd)
+}
+
+func getGatewaysStatus(status []gatewaysStatus) []gatewaysStatus {
+	config, err := getRestConfig(kubeConfig, kubeContext)
+	exitOnError("Error getting REST config for cluster", err)
+
+	submarinerClient, err := submarinerclientset.NewForConfig(config)
+	exitOnError("Unable to get the Submariner client", err)
+
+	existingCfg, err := submarinerClient.SubmarinerV1alpha1().Submariners(OperatorNamespace).Get(submarinercr.SubmarinerName, v1.GetOptions{})
+	if err != nil {
+		exitOnError("error reading from submariner client", err)
+	}
+
+	gateways := existingCfg.Status.Gateways
+	if gateways == nil {
+		exitWithErrorMsg("no gateways found")
+	}
+
+	for _, gateway := range *gateways {
+		haStatus := gateway.Status.HAStatus
+		enpoint := gateway.Status.LocalEndpoint.Hostname
+		totalConnections := len(gateway.Status.Connections)
+		countConnected := 0
+		for _, connection := range gateway.Status.Connections {
+			if connection.Status == submv1.Connected {
+				countConnected += 1
+			}
+		}
+
+		var summary string
+		if totalConnections > 0 && totalConnections == countConnected {
+			summary = "All connections OK"
+		} else {
+			summary = gateway.Status.StatusFailure
+		}
+		connectionString := fmt.Sprintf("%d/%d", countConnected, totalConnections)
```

Comment:
`Status.StatusFailure`  (currently) is populated when there is a failure obtaining the connections in the driver. I think the connection string and the summary somewhat overlap plus if there's partial connectivity then the summary would be empty. I propose we simplify and combine the connection string with the summary:

```suggestion
                if gateway.Status.StatusFailure != "" {
                    summary = gateway.Status.StatusFailure
                } else if totalConnections == 0 {
                    summary = "There are no connections"
                } else if totalConnections == countConnected {
			summary = fmt.Sprintf("All %d connections are established",  totalConnections)
		} else {
			summary = fmt.Sprintf("%d connections out of %d are established",  totalConnections, countConnected)
		}
	
```

Reply:
in case there will be only 1 connection, the summary will show: `All 1 connections are established`

Reply:
yeah good point. How about `All connections (%d) are established`?

Reply:
sounds good.

Reply:
shell I keep the `CONNECTIONS` column? 

```
[maafried@localhost submariner-operator]$ ./bin/subctl show gateways
NODE                HA STATUS       CONNECTIONS     SUMMARY                         
cluster2-worker     active          1/1             All connections (1) are established
```

Reply:
no I think it's redundant.